src/scripts/main.js

```diff
@@ -1,3 +1,236 @@
+    </label>
+    <button>
+      Save to table
+    </button>
+  `;
+
+  return newForm;
+}
+
+function sortTable(index) {
+  const sortedTable = [...tableBody.rows].sort((a, b) => {
+    const valueA = a.children[index].innerText;
+    const valueB = b.children[index].innerText;
+    const sortOrderMultiplier = descendingOrder ? -1 : 1;
+
+    if (index > 2) {
+      return sortOrderMultiplier
+        * (convertToNumber(valueA) - convertToNumber(valueB));
+    }
+
+    return descendingOrder
+      ? valueB.localeCompare(valueA)
+      : valueA.localeCompare(valueB);
+  });
+
+  tableBody.innerHTML = '';
+
+  tableBody.append(...sortedTable);
+  descendingOrder = !descendingOrder;
+}
+
+function convertToNumber(currencyString) {
+  return parseFloat(currencyString.replace(/[$,]/g, ''));
+}
+
+function handleFormSubmit() {
+  const data = new FormData(form);
+  const {
+    name: newName,
+    position,
+    age,
+    office,
+    salary,
+  } = Object.fromEntries(data.entries());
+
+  if (!newName || !age || !position || !salary) {
+    return showNotification(notificationMessage.MainErrorMessage);
+  }
+
+  if (newName.length < 4) {
+    return showNotification(notificationMessage.WrongNameMessage);
+  }
+
+  if (!isNaN(position)) {
+    return showNotification(notificationMessage.WrongPositionMessage);
+  }
+
+  if (age < 18 || age > 90) {
+    return showNotification(notificationMessage.WrongAgeMessage);
+  }
+
+  showNotification(notificationMessage.SuccsessMessage);
+
+  tableBody.insertAdjacentHTML(
+    'afterbegin',
+    `
+      <tr>
+        <td>${newName}</td>
+        <td>${position}</td>
+        <td>${office}</td>
+        <td>${age}</td>
+        <td>$${salary}</td>
+      </tr>
+    `,
+  );
+
+  form.reset();
+}
+
+function showNotification(message) {
+  const { title, description, type } = message;
+  const elementBox = document.createElement('div');
+
+  elementBox.classList.add('notification', type);
+  elementBox.setAttribute('data-qa', 'notification');
+
+  elementBox.innerHTML += `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+
+  document.body.appendChild(elementBox);
+
+  setTimeout(() => {
+    elementBox.remove();
+  }, 2000);
+}
+
+function handeleDoubleClick(target) {
+  const parentRow = target.parentElement;
+  const originalText = target.innerText;
+  const newInput = document.createElement('INPUT');
+
+  newInput.value = originalText;
+  newInput.classList.add('cell-input');
+
+  newInput.setAttribute(
+    'name',
+    tableHeader.rows[0].cells[target.cellIndex].innerText.toLocaleLowerCase(),
+  );
+
+  if (target === parentRow.children[3]) {
+    newInput.setAttribute('type', 'number');
+  }
+
+  if (target === parentRow.children[4]) {
+    newInput.value = convertToNumber(originalText);
+    newInput.setAttribute('type', 'number');
+  }
+
+  target.innerText = '';
+  target.append(newInput);
+  newInput.focus();
+
+  const handleInput = () => {
+    if (newInput.name === 'salary') {
```

Comment:
salary it a magic number
Consider will be better if you create a constants for magic words and use it everywhere